event_handler/event_handler.py

```diff
@@ -46,8 +46,11 @@ def index():
 
     # Verify the signature
     verify_signature = auth_source.verification
-    if not verify_signature(signature, body):
-        raise Exception("Unverified Signature")
+    try:
```

Comment:
try:
  authorized = verify_signature(signature, body)
except Exception as(e):
   

if not authorized:
   abort

Reply:
Yes, that's better. But... thinking about this a bit more... I think I'd like to refactor it slightly, so that we check for the existence of a signature separately from verifying the signature. (We can check for existence in `event_handler.py`, before passing it down to the verification methods.) I think that'll help me debug it, and we can DRY out the code slightly. LMK if you disagree -- thanks!

Reply:
So now it's:

```py
    signature = signature_sources.get(auth_source.signature, None)

    if not signature:
        abort(403, "Signature not found in request headers")

    body = request.data

    # Verify the signature
    verify_signature = auth_source.verification
    if not verify_signature(signature, body):
        abort(403, "Signature does not match expected signature")
```